### ISSUE TITLE
Changelog doesn't contain any intersphinx links

### ISSUE THREAD

**Issue by dstansby** [astropy/astropy#10286]
I was looking down the changelog here: https://docs.astropy.org/en/latest/changelog.html#unreleased, and thought it was a shame that all of the links to new functions/methods/classes didn't have intersphinx links. It would be nice if they were properly linked so if someone sees something they want to take a further look at, they can just click through to the API documentation.

**Comment by bsipocz** [comment]
This is intentional to avoid issues with changing docs references and disappearing API links.

**Comment by bsipocz** [comment]
(It was a constant source of pain of breaking links, if needed I can dig up old references for PRs and issues).

**Comment by dstansby** [comment]
Could one not the links when they disappear? Using the above argument, presumably intersphinx links would never be included anywhere because they might disappear in the future.

**Comment by bsipocz** [comment]
The difference, as I see, is that the CHANGELOG is a static document to snapshot a release, while docs and docstrings are living ones.

**Comment by pllim** [comment]
The decision to not use "live links" in change log was made long ago. I don't see the reason to revert now?

**Comment by dstansby** [comment]
Fair enough, although I think this really does harm discoverability of new features - currently I have to manually do an individual search in the search box for everything I want to know more about, whereas with links I could quickly open a bunch of pages in new tabs. Also, I would have thought new features aren't going to disappear or have their API changed all the time too, so maybe stale links were a bigger issue when the project was less API stable?

**Comment by saimn** [comment]
I think it would be great to have links in the changelog. They shouldn't be used for some specific cases, like deprecation as the functions will disappear later, but other than that it would help a lot to discover new features.

**Comment by pllim** [comment]
I think we should table this until #10334 is resolved.

**Comment by embray** [comment]
Perhaps at least for the most recent release (or even more, any releases for which the docs are available on RTD) it could render the appropriate API links.

**Comment by bsipocz** [comment]
And then remove those API links afterwards? We had those in the past, and I remember the pain of fixing them...

**Comment by dstansby** [comment]
Is it really too much hassle to remove links in a PR that's removing API? I also want to re-iterate my previous point from above:

> Using the above argument, presumably intersphinx links would never be included anywhere because they might disappear in the future and be removed later.

**Comment by Cadair** [comment]
Can't we just ignore broken links in the rendered changelog? Most links working is better than no links working.

**Comment by embray** [comment]
I was thinking this could be done automatically by towncrier but I don't know how easy that would be.  It might be something possible to do within the template used by towncrier.

**Comment by pllim** [comment]
I still don't think this is worth pursuing. I feel like links will be broken again a few years from now and then we get bitten again. This will just spin us around in circles.
